### PR TITLE
CPU & GPU color maps

### PR DESCRIPTION
**Pull request by teh-cmc**
- Just the color maps, both CPU & GPU
- Only polynomial approximations so that we don't end up with different results depending on where the color mapping is done
- Drop the previous Viridis colormap (LUT based) for the reason above
- Color maps are now exposed by `re_renderer` so that everything can benefit without having to import `re_viewer`

_Definitely don't review by on a commit-by-commit basis: they don't make sense, most of them have been salvaged by cherry picking from my gazillon local branches_

PR will follow shortly which integrates these with depth clouds.

### CODE REVIEW

File: crates/re_renderer/src/colormap.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
+#![allow(clippy::excessive_precision)]
```
Comment (teh-cmc) [inline crates/re_renderer/src/colormap.rs:1]: All Vecs in here should be Vec3A/Vec4A, no reason not to

File: crates/re_renderer/src/colormap.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+// NOTE: Keep in sync with `colormap.wgsl`!
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(u32)]
+pub enum ColorMap {
+    Grayscale = 0,
+    ColorMapTurbo = 1,
+    ColorMapViridis = 2,
+    ColorMapPlasma = 3,
+    ColorMapMagma = 4,
+    ColorMapInferno = 5,
```
Comment (teh-cmc) [inline crates/re_renderer/src/colormap.rs:16]: remove those useless prefixes

File: crates/re_renderer/shader/colormap.wgsl [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+// NOTE: Keep in sync with `colormap.rs`!
+const GRAYSCALE:        u32 = 0u;
+const COLORMAP_TURBO:   u32 = 1u;
+const COLORMAP_VIRIDIS: u32 = 2u;
+const COLORMAP_PLASMA:  u32 = 3u;
+const COLORMAP_MAGMA:   u32 = 4u;
+const COLORMAP_INFERNO: u32 = 5u;
+
+fn colormap_srgb(which: u32, t: f32) -> Vec3 {
```
Comment (Wumpf) [inline crates/re_renderer/shader/colormap.wgsl:12]: should document that to is expected to be 0 to 1. (does anything meaningful happen if it isn't?)

Comment (emilk) [reply crates/re_renderer/shader/colormap.wgsl:12]: Should also document the output range (0-1? 0-255?)

File: crates/re_renderer/src/colormap.rs (hunk 2 above)
Comment (teh-cmc) [reply crates/re_renderer/src/colormap.rs:16]: actually don't, I remember why I've put them here now... -.-

File: crates/re_renderer/shader/colormap.wgsl (hunk 3 above)
Comment (teh-cmc) [reply crates/re_renderer/shader/colormap.wgsl:12]: > should document that to is expected to be 0 to 1. (does anything meaningful happen if it isn't?)

I think I'm going to add a `saturate()` everywhere for the GPU ones

**Pull request merged**